README.md

```diff
@@ -76,16 +76,48 @@ class QueryHandler implements IQueryHandler<string, string> {
 }
 ```
 ## Send/request example
-The following example shows how you would send a command and/or request a query into the registrated handlers above.
+The following example shows how you would send a command and/or request a query into the registered handlers above.
 ```TypeScript
 import { Mediator } from 'tsmediator';
 let mediator: Mediator = new Mediator();
 
 mediator.Send(CmdHandler.Type, 5);
 mediator.Request(QueryHandler.Type, 'foo bar');
 ```
+## Middlewares
+The mediator also supports adding middleware to the mediator. This is used if you have any generic actions that you would like to be triggered every time a the mediator is used. What you need to do is create a class that implements `IMediatorMiddleware` and register the middleware with the mediator. 
+
+### Execution order
+The order that you register the middlewares will also be the order that they will be executed.
+If you have registered two middlewares they will be executed in this order: Middleware1 -> Middleware2 -> Handler -> Middleware2 -> Middleware1
+
+### Example
+As an example we can easily create a logging middleware:
+```TypeScript
+class LoggingMiddleware implements IMediatorMiddleware {
+    private logger: ILogger;
+    constructor() {
+        this.logger = new Logger();
+    }
+
+    public PreProcess(payload: any): void {
+        logger.Log(`Processing object ${JSON.stringify(payload)}`)
+    }
+
+    public PostProcess(payload: any, response: any): void {
+        messages.push(`Was: ${JSON.stringify(payload)}, now is: ${JSON.stringify(response)}`);
```

Comment:
Is this supposed to be `logger.Log(`Was: ...`);` instead of `messages.push` or am I missing something? 🤔

Reply:
Nope, you are correct 👍